contracts/IRelayRecipient.sol

```diff
@@ -32,7 +32,20 @@ contract IRelayRecipient {
      *  @param signature - sender's signature over all parameters except approvalData
      *  @param approvalData - extra dapp-specific data (e.g. signature from trusted party)
      */
-    function acceptRelayedCall(address relay, address from, bytes memory encodedFunction, uint gasPrice, uint transactionFee, bytes memory signature, bytes memory approvalData) public view returns (uint);
+     function acceptRelayedCall(
+        address relay,
+        address from,
+        bytes memory encodedFunction,
+        uint256 transactionFee,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        uint256 nonce,
+        bytes memory approvalData,
+        uint256 maxPossibleCharge
+    )
+    public
+    view
+    returns (uint256, bytes memory);
```

Comment:
> At first, I thought it was completely impossible, but then you just did it with revertWithResult

It's relatively easy to do _from Solidity_. But when making an `eth_call` from outside the blockchain, it's impossible with the libraries currently available (see the comments above), which would make testing and interacting really difficult.

Reply:
Also, we could use the `revert()` here in `RelayRecipient#accept`, catch it in `RelayHub#canRelay`, and turn it into a status code. However, we preferred to use the same return value convention in both.

Reply:
I thought in the past to change the resultcode to String instead of uint. the enums are very hard to decipher, and ugly to use since they must be casted to uint...
Re: reading the result from solidity vs external:
we can have `canRelayInternal`, which is used by `relayCall`, and have `canRelay` as a wrapper to call it from the external world.
this way, ACR and friends can `revertWithReason(string)` : relayCall can handle it (and even re-revert with that string, which is helpful if called from ganache)
the `canRelay` will never revert, but return the revert string as a string result code.

Reply:
> I thought in the past to change the resultcode to String instead of uint. the enums are very hard to decipher, and ugly to use since they must be casted to uint...

I wouldn't make this change at this point. We have rolled with enums for quite some time already, and I don't like the idea of introducing this kind of change at this point.

> we can have canRelayInternal, which is used by relayCall, and have canRelay as a wrapper to call it from the external world.

This has the problem of having to disambiguate revert reasons that represent the `PreconditionCheck` values, versus any arbitrary revert reasons that may come from other places in the code. I'm afraid that we would end up with fragile code to try and guess whether a revert reason is a code (which is a good feature for grouping failures) or a random string.